facilitator/src/bin/facilitator.rs

```diff
@@ -709,17 +729,43 @@ fn intake_batch(sub_matches: &ArgMatches) -> Result<(), anyhow::Error> {
         batch_signing_key: batch_signing_key_from_arg(sub_matches)?,
     };
 
+    let batch_id: &str = sub_matches.value_of("batch-id").unwrap();
+    let batch_id: Uuid = Uuid::parse_str(batch_id).unwrap();
```

Comment:
nit: is the explicit `: Uuid` needed here? The compiler can't infer it from the return type of `Uuid::parse_str`? same goes for the `NaiveDateTime::parse_from_str` call just below.

Reply:
I chose to be explicit here to emphasize the change in type for the shadowed variable - from `&str` to `Uuid`. Definitely open to other styles. For instance, we could name the second one `batch_uuid`.

Reply:
You're right, I do like being explicit about shadowing the variable.

Reply:
Oh, funny, I actually looked this up the other day and found it's not yet implemented, then promptly forgot: https://github.com/rust-lang/rust/issues/51114